buildtools/maven/maven_test.go

```diff
@@ -8,30 +8,60 @@ import (
 	"github.com/stretchr/testify/assert"
 
 	"github.com/fossas/fossa-cli/buildtools/maven"
+	"github.com/fossas/fossa-cli/files"
 	"github.com/fossas/fossa-cli/pkg"
 )
 
 var testdataDir = filepath.Join("..", "..", "analyzers", "maven", "testdata")
 
 func TestModules(t *testing.T) {
-	fullPath := filepath.Join(testdataDir, "pom.xml")
+	// Here we mostly just test the discovery of POM files, and analyzers/maven tests that we get the correct
+	// list of MvnModules.
+
+	path1 := filepath.Join(testdataDir, "pom.xml")
 	checked := make(map[string]bool)
-	mods, err := maven.Modules(fullPath, testdataDir, checked)
+	mods, err := maven.Modules(path1, testdataDir, checked)
 	if assert.NoError(t, err) {
 		assert.Len(t, mods, 1)
```

Comment:
Is there a reason you're nesting errors like this? You really should be able to just run the following? I can see why you wouldn't like to be panicking and that's valid although that doesn't concern me as much while running tests.
```
assert.NoError(t, err)
assert.Len(t, mods, 1)

exists, err := files.Exists(mod.Dir, mod.Target)
assert.NoError(t, err)
assert.True(t, exists)
```

Reply:
I guess I've been bitten by panics when not nesting usage of the returned objects within the `if err` check, which only matters if there is in fact an error. I agree that un-indenting will make this look better. Will fix now.